admin/lint_json.m

```diff
@@ -0,0 +1,67 @@
+function lint_json(filesin, outputfile)
+% Use mlint, and convert to a json for easy parsing by WNG
+%
+% Input:
+% ------
+% filesin              char array OR cell array of char arrays detailing files to parse
+%                          if filesin is empty will recurse from current working directory
+% outputfile           char array of filename to write output to (will overwrite)
+%                          if outputfile is empty will write to stdout
+
+    if ~exist('filesin', 'var') || (exist('filesin', 'var') && isempty(filesin))
+        % Default to glob all
+        filesin = {['**', filesep, '*.m']};
+    end
+    if ~exist('outputfile', 'var') || (exist('outputfile', 'var') && isempty(outputfile))
+        fh = 1;
+    else % Open file
+        fh = fopen(outputfile,'w');
+        if fh == -1
+            error("MATLAB:FileOpenError", "Failed to open file %s", outputfile);
+        end
+        cleanup = onCleanup(@()(fclose(fh)));
+    end
+
+    files = [];
+    for i = 1:numel(filesin)
+        flist = dir(filesin{i});
+        % Filter doc files
+        filter = arrayfun(@(x)(startsWith(x.name,'doc_')), flist);
+        filtered = flist(filter);
+        for j = 1:numel(filtered)
+            fprintf("Skipping: %s\n", filtered.name);
+        end
+        flist = flist(~filter);
+        files = [files; arrayfun(@(file)(fullfile(file.folder, file.name)), flist, 'UniformOutput', false)];
+    end
+
+    issues = struct('issues', {{}}, 'size', 0);
+    raw = checkcode(files, '-id');
+    for i = 1:numel(raw)
+        for j = 1:numel(raw{i})
+            raw{i}(j).fileName = files{i};
```

Comment:
as you call `wng_compat` on the next line to parse this object, would it be cleaner to have all modifications done in there, passing `files{i}` as a second argument?